modules/cli/src/main/scala/scala/cli/commands/CommandUtils.scala

```diff
@@ -1,12 +1,43 @@
 package scala.cli.commands
 
+import java.io.File
+import java.nio.file.Paths
+
+import scala.build.Os
+import scala.cli.commands.SetupIde.getProgName
+import scala.util.Try
+
 object CommandUtils {
 
-  def isOutOfDateVersion(newVersion: String, oldVersion: String): Boolean = {
-    import coursier.core.Version
+  /** we only want to check for update within 10% random calls; not everytime
+    */
+  lazy val shouldCheckUpdate: Boolean = {
+    scala.util.Random.nextInt % 10 == 1
+  }
+
+  // Ensure the path to the CLI is absolute
+  lazy val absolutePathToScalaCli: String = {
+
+    val progName = getProgName
```

Comment:
if i use `absolutePathToScalaCli`  in new command, I can get unexpected error `sys.error("setup-ide called in a non-standard way :|")`.

Reply:
I think it is unlikely that you would get the error though, since `setup-side`’s `run` method would not be called in here!

Reply:
Ok, solved by passing `programName` to `getAbsolutePathToScalaCli`